ls1intum/Artemis
# Do not allow instructors to deselect both modes (online code editor and offline IDE) in programming exercises

Fixes #2139 

### Checklist

### Motivation and Context
When creatin a new programming exercise (or editing it), at least one of "Allow Offline IDE" and "Allow Online Editor" must be selected

### Description
Added a validation to prevent deselecting both button "Allow Offline IDE" and "Allow Online Editor". If both of them are deselected, an info icon pops up with a message explaining the problem and both the text and the icon are in red color. Also the "generate" button will be disabled until at least one of the options is selected.

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Administration
3. Select a course (if none possible generate a new one) and go to exercises.
4. Press "Generate a new programming exercise"
5. At the end of the form are the two options"Allow Offline IDE" and "Allow Online Editor" . Try selecting and deselecting both of them and see if the functionality fullfills the conditions. 
6. Test the disabling and enabling of the "generate" button by filling all the mandatory fields. 

### Test Coverage
programming-exercise-update.component.html 100%

### Screenshots
![image](https://user-images.githubusercontent.com/75738365/102811171-40389380-43c5-11eb-8c2d-108fa8f3c097.png)
![image](https://user-images.githubusercontent.com/75738365/102811187-46c70b00-43c5-11eb-854f-237f4355bc6d.png)
![image](https://user-images.githubusercontent.com/75738365/102811385-8db50080-43c5-11eb-97cc-1e67617bf2e2.png)
